chandra_aca/fov_plot.py

```diff
@@ -0,0 +1,309 @@
+    ax.add_patch(box2)
+
+    ax.scatter([-2700, -2700, -2700, -2700, -2700],
+               [2400, 2100, 1800, 1500, 1200],
+               c='orange', edgecolors='none',
+               s=symsize(np.array([10.0, 9.0, 8.0, 7.0, 6.0])))
+
+    [l.set_rotation(90) for l in ax.get_yticklabels()]
+    ax.grid(grid)
+    ax.set_ylabel("Zag (arcsec)")
+    ax.set_xlabel("Yag (arcsec)")
+
+    if quad_bound:
+        pix_range = np.linspace(-510, 510, 50)
+        minus_half_pix = -0.5 * np.ones_like(pix_range)
+        # plot the row = -0.5 line
+        yag, zag = pixels_to_yagzag(minus_half_pix, pix_range)
+        ax.plot(yag, zag, color='magenta', alpha=.4)
+        # plot the col = -0.5 line
+        yag, zag = pixels_to_yagzag(pix_range, minus_half_pix)
+        ax.plot(yag, zag, color='magenta', alpha=.4)
+
+    # plot field if present
+    if stars is not None:
+        if attitude is None:
+            raise ValueError("Must supply attitude to plot field stars")
+        _plot_field_stars(ax, stars, Quaternion.Quat(attitude),
+                          bad_stars=bad_stars, red_mag_lim=red_mag_lim)
+    # plot starcheck catalog
+    if catalog is not None:
+        _plot_catalog_items(ax, catalog)
+    if title is not None:
+        fig.suptitle(title, fontsize='small')
+    return fig
+
+
+def bad_acq_stars(stars):
+    """
+    Return mask of 'bad' stars, by evaluating AGASC star parameters.
+    :param stars: astropy table-compatible set of agasc records of stars
+    :returns: boolean mask true for 'bad' stars
+    """
+    return ((stars['CLASS'] != 0) |
+            (stars['MAG_ACA_ERR'] > 100) |
+            (stars['POS_ERR'] > 3000) |
+            (stars['ASPQ1'] > 0) |
+            (stars['ASPQ2'] > 0) |
+            (stars['ASPQ3'] > 999) |
+            (stars['VAR'] > -9999))
+
+
+def plot_catalog(ra, dec, roll, catalog, starcat_time=DateTime(),
+                 stars=None, bad_stars=None, red_mag_lim=None, title=None):
+    """
+    Make standard catalog plot with a star field and the elements of a catalog.
+
+    :param ra: RA in degrees
+    :param dec: Dec in degrees
+    :param roll: Roll in degrees
+    :param catalog: list of dicts or other astropy.table compatible record structure with
+                    conventional catalog parameters for a set of
+                    ACQ/BOT/GUI/FID/MON items.
+    :param starcat_time: star catalog time.  Used as time for proper motion correction.
+    :param stars: astropy table compatible set of agasc records of stars.  If not supplied,
+                  these will be fetched for the supplied attitude
+    :param bad_stars: mask of stars that should be plotted in BAD_STAR_COLOR
+    :param title: string to be used as suptitle for the figure
+    :returns: matplotlib figure
+    """
+    if stars is None:
+        stars = agasc.get_agasc_cone(ra, dec,
+                                     radius=1.5,
+                                     date=DateTime(starcat_time).date)
```

Comment:
`get_agasc_cone` is written correctly in that it accepts any `DateTime` compatible initializer.  No need to do it here.